core/serialization_v6.c

```diff
@@ -541,40 +657,44 @@ bool serialization_v6_save_shape_as_buffer(const Shape *shape,
         ok = write_preview_chunk_in_buffer(buf + cursor, previewData, previewDataSize, &cursor);
```

Comment:
Before writing preview here, you should be writing `nbShapesCompressedDataSize` into destination buffer `buf` using eg. `write_compressed_chunk_in_buffer` or a similar function